metamask-with/metamask-sdk-nextjs-pages-directory/hooks/useMetaMask.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { type PropsWithChildren } from 'react'
+
+type ConnectAction = { type: 'connect'; wallet: string; balance: string }
+type DisconnectAction = { type: 'disconnect' }
+type PageLoadedAction = {
+  type: 'pageLoaded'
+  isMetaMaskInstalled: boolean
+  wallet: string | null
+  balance: string | null
+}
+type LoadingAction = { type: 'loading' }
+type IdleAction = { type: 'idle' }
+
+type Action =
+  | ConnectAction
+  | DisconnectAction
+  | PageLoadedAction
+  | LoadingAction
+  | IdleAction
+
+type Dispatch = (action: Action) => void
+
+type Status = 'loading' | 'idle' | 'pageNotLoaded'
+
+type State = {
+  wallet: string | null
+  isMetaMaskInstalled: boolean
+  status: Status
+  balance: string | null
+}
+
+const initialState: State = {
+  wallet: null,
+  isMetaMaskInstalled: false,
+  status: 'loading',
+  balance: null,
+} as const
+
+/**
+ * It takes in a state and an action, and returns a new state
+ * @param {State} state - State - the current state of the reducer
+ * @param {Action} action - This is the action that is being dispatched.
+ * @returns The state of the metamask reducer.
+ */
+function metamaskReducer(state: State, action: Action): State {
```

Comment:
Don't hate me, but in lower camel case it think this function per the rules would be: `metaMaskReducer`

There is an argument to be made that you would knock down the entire product name to lowercase as you have done but I don't think it provides a consistent naming convention we can follow as a team and this won't be the last time we run into this one.

As well, below have used: `useMetaMask`

Would love your opinion on this. I know it's knit picky, but... Your code is quite solid I have to pick on something.

Reply:
I would not change code casing for branding considerations. MetaMask is one word, hence it's lowercase. I also think conventions should be flexible but this seems misplaced flexibility.